apis/v1beta1/config_types.go

```diff
@@ -86,15 +86,18 @@ type Extensions struct {
 
 // ConfigSpec defines the desired state of Config
 type ConfigSpec struct {
-	Version              string               `json:"version,omitempty"`
-	Roles                Roles                `json:"roles,omitempty"`
-	UnsupportedOverrides UnsupportedOverrides `json:"unsupportedOverrides,omitempty"`
-	Extensions           Extensions           `json:"extensions,omitempty"`
+	Version              string                `json:"version,omitempty"`
+	Roles                Roles                 `json:"roles,omitempty"`
+	UnsupportedOverrides *UnsupportedOverrides `json:"unsupportedOverrides,omitempty"`
+	Extensions           *Extensions           `json:"extensions,omitempty"`
 }
```

Comment:
...why make these nullable? This is going to cause issues, no?

Reply:
Like there's a bunch of places that will need a null check now that didn't before, even if it all still compiles fine

Reply:
Or we make them nullable or I can close this PR and rethink the whole thing. When  YAML marshalling a ConfigSpec it sets those fields to `{}` and that differs from them not being set, by the user, in the config YAML. Let me know your ideas.